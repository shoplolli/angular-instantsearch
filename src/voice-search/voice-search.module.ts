import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { NgAisVoiceSearch } from './voice-search';

@NgModule({
  declarations: [NgAisVoiceSearch],
  entryComponents: [NgAisVoiceSearch],
  exports: [NgAisVoiceSearch],
  imports: [CommonModule],
})
export class NgAisVoiceSearchModule {}
