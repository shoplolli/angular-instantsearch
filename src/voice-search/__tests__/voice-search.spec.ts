import { createRenderer } from '../../../helpers/test-renderer';
import { NgAisVoiceSearch } from '../voice-search';

const defaultState = {
  isBrowserSupported: true,
  voiceListeningState: {},
};

const render = createRenderer({
  defaultState,
  template: '<ais-voice-search></ais-voice-search>',
  TestedWidget: NgAisVoiceSearch,
});

describe('VoiceSearch', () => {
  describe('button', () => {
    it('calls toggleListening when button is clicked', () => {
      const toggleListening = jest.fn();
      const fixture = render({
        toggleListening,
      });
      const button = fixture.debugElement.nativeElement.querySelector('button');
      button.click();
      expect(toggleListening).toHaveBeenCalledTimes(1);

      button.click(); // see if still clickable when listening
      expect(toggleListening).toHaveBeenCalledTimes(2);
    });
  });

  describe('Rendering', () => {
    it('renders markup without state', () => {
      const fixture = render();
      expect(fixture).toMatchSnapshot();
    });

    it('button disabled in unsupported browser', () => {
      const fixture = render({
        isBrowserSupported: false,
      });
      const button = fixture.debugElement.nativeElement.querySelector('button');
      expect(button.disabled).toBe(true);
    });

    it('with custom template for buttonText (1)', () => {
      const fixture = createRenderer({
        defaultState,
        template: `
          <ais-voice-search>
            <ng-template #button let-isListening="isListening">
              {{ isListening ? 'Stop' : 'Start' }}
            </ng-template>
          </ais-voice-search>
        `,
        TestedWidget: NgAisVoiceSearch,
      })({
        isListening: true,
      });
      const button = fixture.debugElement.nativeElement.querySelector('button');
      expect(button.textContent.trim()).toBe('Stop');
    });

    it('with custom template for buttonText (2)', () => {
      const fixture = createRenderer({
        defaultState,
        template: `
          <ais-voice-search>
            <ng-template #button let-isListening="isListening">
              {{ isListening ? 'Stop' : 'Start' }}
            </ng-template>
          </ais-voice-search>
        `,
        TestedWidget: NgAisVoiceSearch,
      })({
        isListening: false,
      });
      const button = fixture.debugElement.nativeElement.querySelector('button');
      expect(button.textContent.trim()).toBe('Start');
    });

    it('with custom template for status', () => {
      const fixture = createRenderer({
        defaultState,
        template: `
          <ais-voice-search>
            <ng-template #status let-status="status"
                                 let-errorCode="errorCode"
                                 let-isListening="isListening"
                                 let-transcript="transcript"
                                 let-isSpeechFinal="isSpeechFinal"
                                 let-isBrowserSupported="isBrowserSupported">
              <p>status: {{status}}</p>
              <p>errorCode: {{errorCode}}</p>
              <p>isListening: {{isListening}}</p>
              <p>transcript: {{transcript}}</p>
              <p>isSpeechFinal: {{isSpeechFinal}}</p>
              <p>isBrowserSupported: {{isBrowserSupported}}</p>
            </ng-template>
          </ais-voice-search>
        `,
        TestedWidget: NgAisVoiceSearch,
      })({
        isListening: true,
        voiceListeningState: {
          status: 'recognizing',
          transcript: 'Hello',
          isSpeechFinal: false,
          errorCode: undefined,
        },
      });
      expect(
        fixture.debugElement.nativeElement.querySelector(
          '.ais-VoiceSearch-status'
        )
      ).toMatchSnapshot();
    });
  });
});
