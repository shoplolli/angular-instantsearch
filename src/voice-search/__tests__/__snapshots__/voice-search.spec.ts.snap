// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoiceSearch Rendering renders markup without state 1`] = `
<ng-component
  testedWidget={[Function NgAisVoiceSearch]}
>
  <ais-instantsearch>
    <ais-voice-search>
      <div
        class="ais-VoiceSearch"
      >
        <button
          class="ais-VoiceSearch-button"
          disabled=""
          title="Search by voice (not supported on this browser)"
          type="button"
        >
          
          <svg
            fill="none"
            height="16"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            
            <path
              d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
              fill="none"
            />
            <path
              d="M19 10v2a7 7 0 0 1-14 0v-2"
            />
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="23"
            />
            <line
              x1="8"
              x2="16"
              y1="23"
              y2="23"
            />
            
            
          </svg>
        </button>
        <div
          class="ais-VoiceSearch-status"
        >
          
          <p>
            
          </p>
        </div>
      </div>
      
      
    </ais-voice-search>
  </ais-instantsearch>
</ng-component>
`;

exports[`VoiceSearch Rendering with custom template for status 1`] = `
<div
  class="ais-VoiceSearch-status"
>
  
  <p>
    status: recognizing
  </p>
  <p>
    errorCode: 
  </p>
  <p>
    isListening: true
  </p>
  <p>
    transcript: Hello
  </p>
  <p>
    isSpeechFinal: false
  </p>
  <p>
    isBrowserSupported: true
  </p>
</div>
`;
